003_Basic-Example.md

```diff
@@ -0,0 +1,233 @@
+```json
+{
+	"authors": [
+		"username"
+	],
+	"copyright": "Copyright © 2023, username",
+	"dependencies": {
+		"godot-dlang": "~>0.2.0"
+	},
+	"description": "A minimal D application.",
+	"license": "proprietary",
+	"name": "demoextension",
+	"targetName": "demoextension",
+	"targetType": "dynamicLibrary",
+	"targetPath": "../"
+}
+```
+
+Now let's build our first extension.
+Since we've in project root we can't run build from that location, so instead using Visual Studio terminal we will navigate again to our D project and do dub build.
+
+First navigate to D project folder __inside VSCode terminal__.
+```sh
+$ cd demoextension
+```
+
+Now run the build, it **will fail** but that is expected and is normal. 
+We might of course generate bindings beforehand to let it build succesfully, however our intent is to use custom working copy.
+
+```sh
+$ dub build
+```
+
+We need to edit dub.selections.json to tell dub use specific godot-dlang location instead of dub package registry version. 
+It is a package version lock file that fixates used packages versions, normally it should be put under version control system unless you are working on public library.
+Another problem is that we are dealing with local path here, which is complicates work process.
+
+__demoextension/dub.selections.json__: Edit it to something like this
+```json
+{
+	"fileVersion": 1,
+	"versions": {
+		"godot-dlang": { "path": "/home/username/godot-dlang" },
+		"utf_bc": "0.2.1"
+	}
+}
+```
+> Note: on Windows you still have to use forward slash '/' in paths.
+
+And now build it again, this time it should be properly configured and build without error.
+
+```sh
+$ dub build
+```
+
+Done, we should now have our first extension with newly created class ready to use, almost...
+
+## Configuring GDExtension
+
+Wondered why we opened Godot project root folder?
+Now we need create extension description for Godot.
+
+Create a new file inside godot project and fill with following content.
+
+__demoextension.gdextension__:
+```
+[configuration]
+
+entry_symbol = "demoextension_gdextension_entry"
+
+[libraries]
+
+linux.64 = "libdemoextension.so"
+windows.64 = "demoextension.dll"
+```
+
+## Using our Rotator
+
+Go back to godot, if you have it opened all this time you need to reload project using top menu bar `Project->Reload current project` button.
+
+Now create new 2D scene, navigate to file browser panel on the left, drag svg icon on top of empty scene to create new sprite object, select `Icon` object in the scene tree on the left. 
+
+![addsprite.png](media/ch003/addsprite.png)
+
+
+Right click on that `Icon` object an choose attach to new parent, type in Rotator in search bar and confirm.
+
+![reparentnode.png](media/ch003/reparentnode.png)
+
+Move the `Rotator` node a bit away from top left corner, and reset `Icon` transform offset so it will rotate around its center and not some arbitrary point on scene.
+
+Don't forget to save the scene.
+
+Hit play button, and...
+
+Holy sh... 
```

Comment:
Very professional :>

Reply:
It has a nice touch of free spirit of open source movement O_-

We can leave it as is or if it sounds too rude a boring "Oh no" will do.